src/assets/all_assets_request.rs

```diff
@@ -140,9 +124,15 @@ impl AllAssetsRequest {
     /// # Returns
     /// The request object
     /// [AllAssetsRequest](struct.AllAssetsRequest.html)
-    pub fn set_limit(&mut self, limit: u32) -> &mut Self {
-        self.limit = Some(limit);
-        self
+    pub fn set_limit(self, limit: u32) -> Result<AllAssetsRequest, String> {
```

Comment:
We should add the limit boundaries to the documentation, so developers know why their code no longer compiles if it does so.

Reply:
Moved to https://github.com/Baseflow/stellar-rust-sdk/issues/4